core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3533,6 +3533,30 @@ protected RelDataType validateSelectList(
         boolean matches = aggScope.checkAggregateExpr(selectItem, true);
         Util.discard(matches);
       }
+    } else {
```

Comment:
In this else condition isn't there a need to check group-by list is empty ?  I suppose there's an implicit assumption but it would be safer to check.